examples/rbm-python/train_to_print.py

```diff
@@ -0,0 +1,214 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+# (1) update weights
+# (2) load_data row by row?
+
+import cPickle
+import numpy as np
+import numpy.matlib
+import os
+import sys
+
+
+sys.path.append(os.path.join(os.path.dirname(__file__),
+                             '../../build/python'))
+sys.path.append(os.path.join(os.path.dirname(__file__),
+                             '../../build/lib'))
+sys.path.append(os.path.join(os.path.dirname(__file__),'../../build/src'))
+from singa import initializer
+from singa import utils
+from singa import optimizer
+from singa import device
+from singa import tensor
+from singa.proto import core_pb2
+
+
+
+# ??? need normalization??
+def load_train_data(dir_path):
+    file = open('batchdata.csv')
+    traindata = np.genfromtxt(file, delimiter=",")
+    traindata = traindata.astype(np.float32)
+    traindata = traindata.reshape(100, 784, 600) # (batchsize, num_dim, num_batches)
+    return traindata
+
+
+#def load_test_data(dir_path):
+#    images, labels = load_dataset(dir_path + "/test_batch")
+#    return np.array(images,  dtype=np.float32), np.array(labels, dtype=np.int32)
+
+
+def train(data_dir, num_epoch=10, batch_size=100):
+    print 'Start intialization............'
+    weight_lr = 0.1;   # Learning rate for weights 
+    bias_lr = 0.1;   # Learning rate for biases of visible units 
+    weight_decay  = 0.0002;   
+    init_momentum  = 0.5;
+    final_momentum  = 0.9;
+    hdim = 1000
+    vdim = 784
+
+    shape = (vdim, hdim)
+    tweight = tensor.Tensor(shape)
+    initializer.gaussian(tweight, float(0.0), float(0.1)) # ?? is this correct? should have a right hand side?
+#    gweight = np.zeros([vdim, hdim])
+#    gweight = gweight.astype(np.float32) # ?? initialization should be correct
+#    tgweight = tensor.Tensor(gweight.shape)
+#    tgweight.copy_from_numpy(gweight)
+#    vbias = np.zeros(vdim)
+#    vbias = vbias.astype(np.float32)
+#    tvbias = tensor.Tensor(vbias.shape)
+#    tvbias.copy_from_numpy(vbias) # ?? is this copy data from?? what does offset mean?
+#    gvbias = np.zeros(vdim)
+#    gvbias = gvbias.astype(np.float32)
+#    tgvbias = tensor.Tensor(gvbias.shape)
+#    tgvbias.copy_from_numpy(gvbias) # ?? is this copy data from?? what does offset mean?
+#    hbias = np.zeros(hdim)
+#    hbias = hbias.astype(np.float32)
+#    thbias = tensor.Tensor(hbias.shape)
+#    thbias.copy_from_numpy(hbias) # ?? is this copy data from?? what does offset mean?
+#    ghbias = np.zeros(hdim)
+#    ghbias = ghbias.astype(np.float32)
+#    tghbias = tensor.Tensor(ghbias.shape)
+#    tghbias.copy_from_numpy(ghbias) # ?? is this copy data from?? what does offset mean?
+    gweight = np.zeros([vdim, hdim], dtype = np.float32) # ?? initialization correct??
+    tgweight = tensor.from_numpy(gweight)

```

Comment:
you can also merge the above two lines into a single line,

gweight = tensor.from_numpy(np.zeros((vdim, hdim), dtype=np.float32))